scripts/FigS4A-B_celltype-diagnostic-plots.R

```diff
@@ -0,0 +1,196 @@
+source(function_file)
+
+# Create data frame of cell types
+celltype_df <- create_celltype_df(processed_sce)
+
+# Prep SingleR data for plotting -------------------------------------------------------
+
+# extract scores into matrix
+singler_scores <- metadata(processed_sce)$singler_result$scores
+
+# Create data frame for plotting with delta median and the full *non-pruned* cell labels
+delta_median_df <- tibble::tibble(
+  delta_median = rowMaxs(singler_scores) - rowMedians(singler_scores),
+  # Need to grab the non-pruned label for this plot
+  full_labels = metadata(processed_sce)$singler_result$labels,
+  # if pruned.labels are NA ==> low confidence
+  # so, negate for this variable:
+  confident = !is.na(metadata(processed_sce)$singler_result$pruned.labels)
+) |>
+  dplyr::mutate(
+    confident = ifelse(confident, "High-quality", "Low-quality")
+  )
+
+
+# we use inner_join b/c celltype_df does NOT contain "Unknown cell type", which
+#  we do not want to display here
+delta_median_df <- delta_median_df |>
+  dplyr::inner_join(
+    tibble::tibble(
+      full_labels = celltype_df$singler_celltype_ontology,
+      celltype = celltype_df$singler_celltype_annotation
+    ) |> dplyr::distinct()
+  ) |>
+  dplyr::select(-full_labels)
+
+# add column with ordered levels with wrapped labels for visualization
+delta_median_df$annotation_wrapped <- factor(
+  delta_median_df$celltype,
+  # rev() so large groups are at the TOP of the plot
+  levels = rev(levels(delta_median_df$celltype)),
+  labels = rev(stringr::str_wrap(levels(delta_median_df$celltype), 30))
+)
+
+# Subset the data to just confident points for median+/-IQR
+delta_median_confident_df <- delta_median_df |> 
+  dplyr::filter(confident == "High-quality")
+
+# SingleR diagnostic plot ------------------------------------------------------
+
+# Plot delta_median across celltypes colored by pruning
+singler_diagnostic_plot <- ggplot(delta_median_df) +
+  aes(
+    x = delta_median,
+    y = annotation_wrapped,
+    shape = confident,
+    alpha = confident
+  ) +
+  ggforce::geom_sina(
+    size = 0.8,
+    color = "black", # will get applied to all confident points and non-confident outline
+    fill = "white", # will apply to non-confident fill only
+    position = position_dodge(width = 0.05) # Keep both types of points mostly in line
+  ) +
+  # Handle points aesthetics:
+  #  confident are closed black with alpha = 0.5
+  #  not confident are open black with alpha = 1
+  scale_shape_manual(values = c(19, 21)) +
+  scale_alpha_manual(values = c(0.5, 1)) +
+  labs(
+    x = "Delta median statistic",
+    y = "Cell type annotation",
+    shape = "Cell type annotation quality"
+  ) +
+  # add median diamond for confident points only
+  stat_summary(
+    data = delta_median_confident_df,
+    color = "red",
+    geom = "point",
+    fun = "median",
+    shape = 18,
+    size = 2.25,
+    alpha = 0.9
+  ) +
+  guides(
+    alpha = FALSE,
```

Comment:
ggplot2 is giving me a warning here (probably a thing we missed in the qc report b/c warnings may have been off for the chunk?)

```
Warning message:
The `<scale>` argument of `guides()` cannot be `FALSE`. Use "none" instead as
of ggplot2 3.3.4. 
```